rpcserver.go

```diff
@@ -2148,6 +2150,66 @@ func handleGetHashesPerSec(s *rpcServer, cmd interface{}, closeChan <-chan struc
 	return int64(s.server.cpuMiner.HashesPerSecond()), nil
 }
 
+// handleGetHeaders implements the getheaders command.
+//
+// NOTE: This is a btcsuite extension ported from
+// github.com/decred/dcrd.
+func handleGetHeaders(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*btcjson.GetHeadersCmd)
+	pBlockLocators := make([]*chainhash.Hash, len(c.BlockLocators))
+	for i := range c.BlockLocators {
+		pBlockLocator, err := chainhash.NewHashFromStr(c.BlockLocators[i])
+		if err != nil {
+			return nil, &btcjson.RPCError{
+				Code: btcjson.ErrRPCInvalidParameter,
+				Message: "Failed to decode block locator: " +
+					err.Error(),
+			}
+		}
+		pBlockLocators[i] = pBlockLocator
+	}
+	if c.HashStop != "" {
+		c.HashStop = strings.Repeat("0", chainhash.HashSize*2)
```

Comment:
ACK. Created `chainhash.Decode` function and updated `chainhash.NewHashFromStr` and `handleGetHeaders` to use it like dcrd does.